sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageBatch.cs

```diff
@@ -118,15 +118,15 @@ public override bool TryAddMessage(ServiceBusMessage message)
                         ? OverheadBytesSmallMessage
                         : OverheadBytesLargeMessage);
 
-                if (size > MaxSizeInBytes)
+                if (size > MaxSizeInBytes || BatchMessages.Count >= Options.MaxSizeInNumberOfItems.Value)
```

Comment:
Wasn't entirely sure if you would prefer to copy the value to a field as done for

```
MaxSizeInBytes = options.MaxSizeInBytes.Value;
```

given that we also store the options locally I went with accessing the options directly